### PR TITLE
Support attribution scope fields in source and trigger.

### PR DESCRIPTION
**Pull request by feifeiji89**
- Source fields:
         -- attribution scope limit
         -- attribution scopes
         -- max num view states
- Trigger field: attribution scope

***
<a href="https://pr-preview.s3.amazonaws.com/feifeiji89/attribution-reporting-api/pull/1197.html" title="Last updated on Mar 18, 2024, 5:28 PM UTC (f476b8c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/1197/d991bb9...feifeiji89:f476b8c.html" title="Last updated on Mar 18, 2024, 5:28 PM UTC (f476b8c)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -806,6 +806,12 @@ An attribution source is a [=struct=] with the following items:
 :: A [=trigger-data matching mode=].
 : <dfn>debug cookie set</dfn> (default false)
 :: A [=boolean=].
+: <dfn>attribution scope limit</dfn>
+:: Null or a non-negative 32-bit integer, number of distinct [=attribution source/attribution scopes=] allowed per [=attribution source/reporting origin=].
```
Comment (giladbarkan-github) [inline index.bs:810]: What does a limit of zero mean?

File: index.bs [hunk 2]
```diff
@@ -806,6 +806,12 @@ An attribution source is a [=struct=] with the following items:
...
+: <dfn>attribution scopes</dfn>
+:: Null or a list of non-negative 32-bit integers.
```
Comment (giladbarkan-github) [inline index.bs:812]: Can't scopes be negative (like priority)?

File: index.bs [hunk 3]
```diff
@@ -931,6 +937,8 @@ An attribution trigger is a [=struct=] with the following items:
 :: An [=aggregatable source registration time configuration=].
 : <dfn>trigger context ID</dfn>
 :: Null or a [=string=].
+: <dfn>attribution scope</dfn>
```
Comment (apasel422) [inline index.bs:940]: This field is never set. Don't we have to modify parsing?

File: index.bs [hunk 4]
```diff
@@ -806,6 +806,12 @@ An attribution source is a [=struct=] with the following items:
 :: A [=trigger-data matching mode=].
 : <dfn>debug cookie set</dfn> (default false)
 :: A [=boolean=].
+: <dfn>attribution scope limit</dfn>
+:: Null or a non-negative 32-bit integer, number of distinct [=attribution source/attribution scopes=] allowed per [=attribution source/reporting origin=].
+: <dfn>attribution scopes</dfn>
+:: Null or a list of non-negative 32-bit integers.
+: <dfn>max number of view states</dfn>
+:: Null or a non-negative 32-bit integer, maximum number view states that an API caller plans to use.
```
Comment (apasel422) [inline index.bs:814]: Please clarify what a "view state" is. That term doesn't appear elsewhere in the specification.

File: index.bs [hunk 5]
```diff
@@ -806,6 +806,12 @@ An attribution source is a [=struct=] with the following items:
 :: A [=trigger-data matching mode=].
 : <dfn>debug cookie set</dfn> (default false)
 :: A [=boolean=].
+: <dfn>attribution scope limit</dfn>
```
Comment (apasel422) [inline index.bs:809]: Where is this limit enforced?

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: index.bs (hunk 2 above)
Comment (apasel422) [inline index.bs:812]: Why does this need to be nullable? How is null different from an empty list?

Comment (linnan-github) [inline index.bs:812]: Should this be a set to avoid duplicates?

**Pull request closed without merging**